cpp/src/phonenumbers/phonenumberutil.cc

```diff
@@ -1527,12 +1525,76 @@ void PhoneNumberUtil::FormatNsnWithCarrier(const string& number,
       : metadata.intl_number_format();
   const NumberFormat* formatting_pattern =
       ChooseFormattingPatternForNumber(available_formats, number);
+  NumberFormat new_format;
+  bool consistent_optional_rule = TRUE;
+  bool consistent_fg_rule = TRUE;
   if (!formatting_pattern) {
-    formatted_number->assign(number);
-  } else {
-    FormatNsnUsingPatternWithCarrier(number, *formatting_pattern, number_format,
-                                     carrier_code, formatted_number);
+      // Set the pattern to the possible national patterns, check the number
```

Comment:
I'd make this a bit simpler by just saying "check the number is possible, otherwise don't try to do any formatting".